File: 025.go
package main

import ("math"; "fmt")

var fibNumList map[int]uint64

func fib(n int) uint64 {
    fibNumCached, exists := fibNumList[n]
    if exists {
        return fibNumCached
    }
    if n <= 2 {
        fibNumList[n] = 1
        return 1
    }
    res := fib(n-1) + fib(n-2)
    fibNumList[n] = res
    return res
}

func main() {
    fibNumList = make(map[int]uint64)
    fibNumList[1] = 1
    fibNumList[2] = 1
    nDigits := 1000
    limit := uint64(math.Pow(10, float64(nDigits) - 1)) - 1    
    fmt.Printf("Límite superior para el FibNum: %d\n", limit)
    index := 1
    for {
        num := fib(index)
        if num > limit {
            fmt.Printf("F(%d) = %d\n", index, num)
            return
        }
        index++
    } 
}

File: 025.py

fib_numbers_cache = {}

def fib(num):
	if num in fib_numbers_cache:
		return fib_numbers_cache[num]
	if num <= 2:
		fibNum = 1
	else:
		fibNum = fib(num - 1) + fib(num - 2)
	fib_numbers_cache[num] = fibNum
	return fibNum

n = 1
n_digits = 1000
fib_num_max = 10**(n_digits - 1) - 1
while True:
	fn = fib(n)
	if fn > fib_num_max:
		print "END: ", n, "=>", fn
		break
	n += 1

